### ISSUE TITLE
Public Link with Search or Advanced Search Options for Entire TREE [Trilium Server]

### ISSUE THREAD

**Issue by mangoose20026** [zadam/trilium#4816]
### Describe feature

Public Link with Search or Advanced Search Options for Entire TREE [Trilium Server]

The public link generated for shared notes (Entire TREE) in Trilium do not inherently include advanced search functionalities. The shared view primarily allows for browsing through the notes and their content. Users can navigate through the hierarchy, but they might not have access to advanced search features unless they are using the full application with appropriate permissions.

Either Search and Advanced search is needed or there should be someway to have guest or read-only credentials with search and advanced search for shared user

### Additional Information

Basically provide advanced search and search for entire tree collection to public or anyone to read them

**Comment by rauenzi** [comment]
I added a search API for share pages in #4298 but the default share template has not been updated to make use of it yet. I am working with Zadam to integrate a new default share template that will use it. However if you want to have searching before that's integrated you can implement it yourself via share templates.

If you're interested in that method, you can see it in action here: https://trilium.rocks. It's a website entirely run by Trilium's share feature using a custom share template that includes several features the current default share template does not most relevantly, search. The implementation of the template and the corresponding JavaScript can be found in the repo (https://github.com/rauenzi/trilium.rocks) for the website, and if you just want to use that template directly, I have an article on it here: https://trilium.rocks/xecUJ8eL3pvM

**Comment by mangoose20026** [comment]
@rauenzi Awesome...Please integrate it and that would solve all issues.

**Comment by eliandoran** [comment]
There is a basic support for search in share, as stated by @zerebos .
We don't plan to invest more time currently in improving the search, but PRs are welcome as long as they don't change the share functionality into something it isn't.
